connectors/src/connectors/zendesk/lib/permissions.ts

```diff
@@ -104,3 +83,179 @@ export async function retrieveSelectedNodes({
+            parentInternalId: parentInternalId,
+            type: "folder",
+            title: "Help Center",
+            sourceUrl: null,
+            expandable: true,
+            permission: "none",
+            dustDocumentId: null,
+            lastUpdatedAt: null,
+          };
+          nodes.push(helpCenterNode);
+        }
+        break;
+      }
+      // If the parent is a brand's tickets, we retrieve the list of tickets for the brand.
+      case "tickets": {
+        if (filterPermission === "read") {
+          const ticketsInDb = await ZendeskBrandResource.fetchReadOnlyTickets({
+            connectorId,
+            brandId: objectId,
+          });
+          nodes = ticketsInDb.map((ticket) =>
+            ticket.toContentNode({ connectorId })
+          );
+        }
+        break;
+      }
+      // If the parent is a brand's help center, we retrieve the list of Categories for this brand.
+      case "help-center": {
+        const categoriesInDatabase =
+          await ZendeskBrandResource.fetchReadOnlyCategories({
+            connectorId,
+            brandId: objectId,
+          });
+        if (isReadPermissionsOnly) {
+          nodes = categoriesInDatabase.map((category) =>
+            category.toContentNode({ connectorId })
+          );
+        } else {
+          await changeZendeskClientSubdomain({
+            client: zendeskApiClient,
+            brandId: objectId,
+          });
+          const categories =
+            await zendeskApiClient.helpcenter.categories.list();
+          nodes = categories.map((category) => {
+            const matchingDbEntry = categoriesInDatabase.find(
+              (c) => c.categoryId === category.id
+            );
+            return {
+              provider: connector.type,
+              internalId: getCategoryInternalId(connectorId, category.id),
+              parentInternalId: parentInternalId,
+              type: "folder",
+              title: category.name,
+              sourceUrl: category.html_url,
+
+              expandable: false,
+              permission: matchingDbEntry ? "read" : "none",
+              dustDocumentId: null,
+              lastUpdatedAt: matchingDbEntry?.updatedAt.getTime() ?? null,
+            };
+          });
+        }
+        break;
+      }
+      // If the parent is a category, we retrieve the list of articles for this category.
+      case "category": {
+        if (isReadPermissionsOnly) {
+          const articlesInDb =
+            await ZendeskCategoryResource.fetchReadOnlyArticles({
+              connectorId,
+              categoryId: objectId,
+            });
+          nodes = articlesInDb.map((article) =>
+            article.toContentNode({ connectorId })
+          );
+        }
+        break;
+      }
+      // Single tickets and articles have no children.
+      case "ticket":
+      case "article":
+      case null:
+        return [];
+    }
```

Comment:
ends your switch case with a `default: assertNever(type)` - this will ensure we don't forget to handle new cases in case we add a new value to the enum